pkg/ccl/backupccl/create_scheduled_backup_test.go

```diff
@@ -0,0 +1,328 @@
+			require.Equal(t, ScheduledBackupExecutionArgs_FULL, arg.BackupType)
+			require.Equal(t, tc.backupStmt, arg.BackupStatement)
+		})
+	}
+}
+
+func TestScheduleBackup(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	th, cleanup := newTestHelper(t)
+	defer cleanup()
+
+	th.sqlDB.Exec(t, `
+CREATE DATABASE db;
+USE db;
+
+CREATE TABLE t1(a int);
+INSERT INTO t1 values (1), (10), (100);
+
+CREATE TABLE t2(b int);
+INSERT INTO t2 VALUES (3), (2), (1);
+
+CREATE TABLE t3(c int);
+INSERT INTO t3 VALUES (5), (5), (7);
+
+CREATE DATABASE other_db;
+USE other_db;
+
+CREATE TABLE t1(a int);
+INSERT INTO t1 values (-1), (10), (-100);
+`)
+
+	// We'll be manipulating schedule time via th.env, but we can't fool actual backup
+	// when it comes to AsOf time.  So, override AsOf backup clause to be the current time.
+	th.cfg.TestingKnobs.(*jobs.TestingKnobs).OverrideAsOfClause = func(clause *tree.AsOfClause) {
+		expr, err := tree.MakeDTimestampTZ(th.cfg.DB.Clock().PhysicalTime(), time.Microsecond)
+		require.NoError(t, err)
+		clause.Expr = expr
+	}
+
+	testCases := []struct {
+		name         string
+		schedule     string
+		restore      string
+		verifyTables map[string]string
+	}{
+		{
+			name:         "cluster-backup",
+			schedule:     "CREATE SCHEDULE FOR BACKUP TO $1 RECURRING '@hourly'",
+			restore:      "RESTORE db.t1 FROM $1 WITH into_db=restore_db",
+			verifyTables: map[string]string{"t1": "db.t1"},
+		},
+		{
+			name:         "tables-backup-with-encryption",
+			schedule:     "CREATE SCHEDULE FOR BACKUP db.t2, db.t3 TO $1 WITH encryption_passphrase='secret' RECURRING '@hourly'",
+			restore:      "RESTORE TABLE db.t2, db.t3 FROM $1 WITH encryption_passphrase='secret', into_db=restore_db",
+			verifyTables: map[string]string{"t2": "db.t2", "t3": "db.t3"},
+		},
+		{
+			name:         "table-backup-in-different-dbs",
+			schedule:     "CREATE SCHEDULE FOR BACKUP db.t1, other_db.t1, db.t3 TO $1 RECURRING '@hourly'",
+			restore:      "RESTORE TABLE db.t3, other_db.t1 FROM $1 WITH into_db=restore_db",
+			verifyTables: map[string]string{"t1": "other_db.t1", "t3": "db.t3"},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			destination := "nodelocal://0/backup/" + tc.name
+			schedules, err := th.createBackupSchedule(tc.schedule, destination)
+			require.NoError(t, err)
+			require.EqualValues(t, 1, len(schedules))
+
+			// Force the schedule to execute.
+			th.env.SetTime(schedules[0].NextRun().Add(time.Second))
+			require.NoError(t,
+				th.cfg.DB.Txn(context.Background(), func(ctx context.Context, txn *kv.Txn) error {
+					return th.executeSchedules(ctx, allSchedules, txn)
+				}))
+
+			// Wait for the backup to complete.
+			th.waitForSuccessfulScheduledJob(t, schedules[0].ScheduleID())
+
+			// Verify backup.
+			th.sqlDB.Exec(t, "CREATE DATABASE restore_db; USE restore_db")
+			defer th.sqlDB.Exec(t, "DROP DATABASE restore_db")
+			th.sqlDB.Exec(t, tc.restore, destination)
```

Comment:
We could just run `SELECT table_name FROM [SHOW BACKUP $1]` and verify we get back the set of verify tables -- we only need to be testing that the schedule backed up what it was supposed to backup, not that backup and restore work since that's the responsibility of the backup tests.